content/en/blog/_posts/2019-06-13-volume-cloning-alpha.md

```diff
@@ -0,0 +1,114 @@
+
+Kubernetes supports three types of volume plugins: in-tree, Flex, and CSI. See [Kubernetes Volume Plugin FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md) for details.
+
+Cloning is only supported for CSI drivers (not for in-tree or Flex). To use the Kubernetes cloning feature, ensure that a CSI Driver that implements cloning is deployed on your cluster.
+
+## Kubernetes API and Cloning
+
+The cloning feature in Kubernetes is an enhancement to the existing PersistentVolumeClaim objects and it's API.  There are no new objects introduced to enable cloning, instead the existing DataSource field in the PersistentVolumeClaim object is expanded to be able to accept the name of an existin PersistentVolumeClaim in the users namespace.  It is important to note that from a users perspective a clone is just another Volume, the only difference being that that volume is being populated with the contents of another volume at creation time.  After creation it behaves exactly like any other Kubernetes Volume and adhers to the same behaviors and rules as any other Volume.
+
+## Kubernetes Cloning Requirements
+
+Before using Kubernetes Volume Cloning, you must:
+
+* Ensure a CSI driver implementing Cloning is deployed and running on your Kubernetes cluster.
+* Enable the Kubernetes Volume Cloning feature via new Kubernetes feature gate (disabled by default for alpha):
+  * Set the following flag on the API server binary: `--feature-gates=VolumePVCDataSource=true`
+
+## Creating a clone with Kubernetes
+
+To provision a new volume pre-populated with data from an existing Kubernetes Volume, use the dataSource field in the `PersistentVolumeClaim`.  There are three parameters:
+
+* name - name of the `PersistentVolumeClaim` object to use as source
+* kind - must be `PersistentVolumeClaim`
+* apiGroup - must be `""`
+
+The namespace of the source `PersistentVolumeClaim` (source) object is assumed to be the same as the namespace of the new `PersistentVolumeClaim` (dstination) being created object.
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-clone
+  Namespace: demo-namespace
+spec:
+  storageClassName: csi-storageclass
+  dataSource:
+    name: src-pvc
+    kind: PersistentVolumeclaim 
+    apiGroup: ""
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+When the `PersistentVolumeClaim` object is created, it will trigger provisioning of a new volume that is pre-populated with data from the specified `dataSource` volume.  It is the sole responsbility of the CSI Plugin to implement the cloning of volumes.
+
+## As a storage vendor, how do I add support for cloning to my CSI driver?
+
+To implement the cloning feature, a CSI driver MUST add cloning support to its controller capabilities `CLONE_VOLUME`. For details, see [the CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md).
```

Comment:
Most of this information should probably just go into kubernetes-csi docs page, and the blog post can point to that page.